client/my-sites/stats/hooks/use-should-gate-stats.ts

```diff
@@ -128,12 +130,18 @@ export const shouldGateStats = ( state: object, siteId: number | null, statType:
 		}
 
 		const isSiteCommercial = getSiteOption( state, siteId, 'is_commercial' ) || false;
-		// Paywall basic stats for commercial sites.
 		if ( isSiteCommercial ) {
-			return [
-				...jetpackStatsCommercialPaywall,
-				...granularControlForJetpackStatsCommercialPaywall,
-			].includes( statType );
+			const billableMonthlyViews = getPlanUsageBillableMonthlyViews( state, siteId );
+			// Paywall basic stats for commercial sites with monthly views of more than 1k.
+			if ( billableMonthlyViews >= MIN_MONTHLY_VIEWS_TO_APPLY_PAYWALL ) {
+				return [
+					...jetpackStatsCommercialPaywall,
+					...granularControlForJetpackStatsCommercialPaywall,
+				].includes( statType );
+			}
+
+			// Paywall advanced stats for commercial sites with monthly views less than 1k.
+			return [ ...jetpackStatsAdvancedPaywall ].includes( statType );
 		}
 
 		// Paywall advanced stats for non-commercial sites.
```

Comment:
L152-154 would be applying the paywall to dotcom sites, wouldn't it? I think we should roll back the change 🤔

Reply:
Ah right I got it, dotcom sites wouldn't be able to see these modules yet, so it should be sweet.

Reply:
> dotcom sites wouldn't be able to see these modules yet

Yes, and I am leaving the check to prevent advanced modules (UTM, Devices for now) from showing on the WPCOM sites without Stats commercial purchase someday.